packages/spectral/src/types/Inputs.ts

```diff
@@ -109,102 +109,110 @@ interface BaseInputField {
   required?: boolean;
 }
 
-export interface StringInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "string";
-  /** Collection type of the InputField */
-  collection?: InputFieldCollection;
-  /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
-}
+type CollectionOptions<T> =
+  | SingleValue<T>
+  | ValueListCollection<T>
+  | KeyValueListCollection<T>;
 
-export interface DataInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "data";
+interface SingleValue<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection?: undefined;
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: T;
 }
 
-export interface TextInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "text";
+interface ValueListCollection<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection: "valuelist";
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: T[];
 }
 
-export interface PasswordInputField extends BaseInputField {
-  /** Data type the InputField will collect. */
-  type: "password";
+interface KeyValueListCollection<T> {
   /** Collection type of the InputField */
-  collection?: InputFieldCollection;
+  collection: "keyvaluelist";
   /** Default value for this field. */
-  default?: unknown;
-  /** Dictates possible choices for the input. */
-  model?: InputFieldChoice[];
-  /** Clean function */
-  clean?: InputCleanFunction<this["default"]>;
+  default?: { key: string; value: T }[];
```

Comment:
It would probably be most proper to use the `interface KeyValuePair<V = unknown>` type here.